### ISSUE TITLE
cpp: space added BEFORE reference & for a varible with a namespace qualifier when sp_arith = true

### ISSUE THREAD

**Issue by JodiTheTigger** [uncrustify/uncrustify#3865]
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastebin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

Version:
```
Uncrustify-0.75.0-358-a62f15bef
```

Config
```
sp_arith = add
```

Input
```cpp
MOCK_METHOD(bool, isSelectable, (const std::string& name), (override));
MOCK_METHOD(bool, isSelectable, (std::string& name), (override));
MOCK_METHOD(bool, isSelectable, std::string& name, (override));
MOCK_METHOD(bool, isSelectable, const std::string& name, (override));
Rate(const std::string& rate);
```

Expected: Unchanged

Output:
```cpp
MOCK_METHOD(bool, isSelectable, (const std::string & name), (override));
MOCK_METHOD(bool, isSelectable, (std::string & name), (override));
MOCK_METHOD(bool, isSelectable, std::string & name, (override));
MOCK_METHOD(bool, isSelectable, const std::string & name, (override));
Rate(const std::string & rate);
```

regression from 0.73 found via git bisect:
```
commit 9ad84db68e9c9487f77637fd5f16fc7f330b7c2d (fix/2103)
Author: Guy Maurel <guy.j@maurel.de>
Date:   Wed Jan 5 16:49:53 2022 +0100

    fixes #2103

```

**Comment by JodiTheTigger** [comment]
Looking at the regression at 9ad84db68e9c9487f77637fd5f16fc7f330b7c2d - this seems like it's a no-win situation, as there doesn't seem to be a way to tell the difference between an enum, and a typename with a namespace qualifier.

Not sure of the correct solution. Ideas are:
* Revert the change - and reintroduce the enum issue
* Change `sp_arith` to have a third option: `On, off, everything_except_enums_with_namespace`
* Add checks so this doesn't happen if inside `PCF_IN_FCN_CALL` or `PCF_IN_FCN_CTOR` or `PCF_IN_FCN_DEF`
  * But that'll probably break code when calling a function with a valid `enum & enum::value` sub expression
* Add another config option?
* Something else?

Waiting on guidance for a fix to attempt

**Comment by micheleCTDEAdmin** [comment]
I think option 3 makes sense. The regression you have reported is inside function call argument list, so a & should always be a reference there.

**Comment by JodiTheTigger** [comment]
Ok initial fix didn't work - I'll try again tomorrow.

**Comment by JodiTheTigger** [comment]
Found a bunch more edge cases (might affect `*` as well). So I'll take a few days to get them all sorted.

**Comment by JodiTheTigger** [comment]
I'll raise the `*` based regressions as a separate issue so I can at least get the `&` fixes done.

**Comment by JodiTheTigger** [comment]
Ok, I have fixes for three 0.73 regressions for `&`: https://github.com/uncrustify/uncrustify/pull/3900

**Comment by JodiTheTigger** [comment]
Posted issue with `*` here: https://github.com/uncrustify/uncrustify/issues/3904
